### PR TITLE
Check if content-encoding is being stored as object metadata on s3 put

### PR DESCRIPTION
**Pull request by mashetty330**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2054074
Signed-off-by: mashetty <mashetty@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/bucket_utils.py [hunk 1]
```diff
@@ -937,7 +939,11 @@ def s3_put_object(s3_obj, bucketname, object_key, data, content_type=""):
 
     """
     return s3_obj.s3_client.put_object(
```
Comment (parikshithb) [inline ocs_ci/ocs/bucket_utils.py:941]: Verify whether this change does not cause any failures of tests using  s3_obj.s3_client.put_object(

File: tests/manage/mcg/test_write_to_bucket.py [hunk 2]
```diff
@@ -360,3 +364,39 @@ def test_write_to_bucket_rbd_cephfs(
                     awscli_pod_session, target_dir, full_object_path, mcg_obj
                 )
             )
+
+    @tier2
+    @bugzilla("2054074")
+    def test_content_encoding_with_write(self, bucket_factory, mcg_obj_session):
```
Comment (parikshithb) [inline tests/manage/mcg/test_write_to_bucket.py:370]: Doc text regarding the test test

File: tests/manage/mcg/test_write_to_bucket.py [hunk 3]
```diff
@@ -360,3 +364,39 @@ def test_write_to_bucket_rbd_cephfs(
                     awscli_pod_session, target_dir, full_object_path, mcg_obj
                 )
             )
+
+    @tier2
+    @bugzilla("2054074")
```
Comment (parikshithb) [inline tests/manage/mcg/test_write_to_bucket.py:369]: add skip if version if not supported in versions others than fixed in version of bz

File: tests/manage/mcg/test_write_to_bucket.py [hunk 4]
```diff
@@ -360,3 +364,39 @@ def test_write_to_bucket_rbd_cephfs(
...
+    def test_content_encoding_with_write(self, bucket_factory, mcg_obj_session):
+        # create bucket
+        bucket_name = bucket_factory()[0].name
+        logger.info(f"Bucket created {bucket_name}")
+
+        # create a random file and then zip it
+        with open("randomFile", "wb") as f:
```
Comment (parikshithb) [inline tests/manage/mcg/test_write_to_bucket.py:376]: provide a random generated file name string

**Review by parikshithb — COMMENTED** [review]
Few comments, LG,

File: ocs_ci/ocs/bucket_utils.py (hunk 1 above)
Comment (mashetty330) [reply ocs_ci/ocs/bucket_utils.py:941]: ack

File: tests/manage/mcg/test_write_to_bucket.py (hunk 2 above)
Comment (mashetty330) [reply tests/manage/mcg/test_write_to_bucket.py:376]: ack

File: tests/manage/mcg/test_write_to_bucket.py (hunk 3 above)
Comment (mashetty330) [reply tests/manage/mcg/test_write_to_bucket.py:374]: ack

File: tests/manage/mcg/test_write_to_bucket.py (hunk 4 above)
Comment (mashetty330) [reply tests/manage/mcg/test_write_to_bucket.py]: ack

File: tests/manage/mcg/test_write_to_bucket.py [hunk 5]
```diff
@@ -360,3 +369,46 @@ def test_write_to_bucket_rbd_cephfs(
...
+        head_obj = s3_head_object(
+            s3_obj=mcg_obj_session, bucketname=bucket_name, object_key="random.zip"
+        )
+        assert (
+            head_obj["ContentEncoding"] == "zip"
+        ), "Put object operation doesn't store ContentEncoding!!"
+
+        # cleanup the files created
+        os.remove(f"{filename}.zip")
+        os.remove(f"{filename}")
```
Comment (PrasadDesala) [inline tests/manage/mcg/test_write_to_bucket.py:414]: Should this be in teardown? If the test breaks before reaching here, the files will left uncleaned

File: tests/manage/mcg/test_write_to_bucket.py [hunk 6]
```diff
@@ -360,3 +369,46 @@ def test_write_to_bucket_rbd_cephfs(
...
+        bucket_name = bucket_factory()[0].name
+        logger.info(f"Bucket created {bucket_name}")
+
+        # create a random file and then zip it
+        filename = f"random-{uuid4().hex}"
+        with open(filename, "wb") as f:
+            f.write(os.urandom(1000))
+        zip_obj = ZipFile(f"{filename}.zip", "w")
+        zip_obj.write(f"{filename}", compress_type=zipfile.ZIP_DEFLATED)
+        zip_obj.close()
```
Comment (PrasadDesala) [inline tests/manage/mcg/test_write_to_bucket.py:392]: You can change it to below and spare the use of close
with ZipFile(f"{filename}.zip", "w") as zip:
    zip.write(f"{filename}", compress_type=zipfile.ZIP_DEFLATED)

File: tests/manage/mcg/test_write_to_bucket.py (hunk 5 above)
Comment (mashetty330) [reply tests/manage/mcg/test_write_to_bucket.py:414]: Good suggestion! Its better to have it that way

**Review by PrasadDesala — APPROVED** [review]

**Review by parikshithb — APPROVED** [review]

**Pull request merged**